whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapElasticFullTextSearchService.java

```diff
@@ -90,19 +91,34 @@ protected List<RpslObject> doSearch() throws IOException {
                     return results;
                 }
 
-                private QueryBuilder getQueryBuilder(final String[] fields, final String term) {
-                    if (term.indexOf('*') == -1 && term.indexOf('?') == -1) {
-                        final MultiMatchQueryBuilder multiMatchQuery = new MultiMatchQueryBuilder(term, fields)
-                                .type(MultiMatchQueryBuilder.Type.PHRASE_PREFIX)
-                                .operator(Operator.AND);
-                        return multiMatchQuery;
+                private QueryBuilder getQueryBuilder(final String[] fields, final String term, final boolean matchExact) {
+                    if (hasNotWildCard()) {
+                        return matchExact ? createExactMatchQuery() :
+                                new MultiMatchQueryBuilder(term, fields)
+                                        .type(MultiMatchQueryBuilder.Type.PHRASE_PREFIX)
+                                        .operator(Operator.AND);
                     }
+                    return createWildCardQuery();
+                }
+
+                private boolean hasNotWildCard(){
+                    return term.indexOf('*') == -1 && term.indexOf('?') == -1;
+                }
+
+                private BoolQueryBuilder createExactMatchQuery(){
+                    final BoolQueryBuilder rawSearchBuilder = QueryBuilders.boolQuery();
+                    for (String field : fields) {
+                        rawSearchBuilder.should(QueryBuilders.termQuery(String.format("%s.raw", field), term));
```

Comment:
termQuery: 
looks for the exact term 
It does not analyze the input term or the field's value. Instead, it checks for an exact match between the provided value (term) and the value stored in the field.